### ISSUE TITLE
Connection error ERR_INVALID_ARG_TYPE

### ISSUE THREAD

**Issue by rclabs** [eXigentCoder/krpc-node#16]
KRPC version: 0.4.8

I am seeing a connection error trying to run the following program:

    'use strict';
    let { createClient, spaceCenter } = require('krpc-node');

    async function main() {
        const client = await createClient();
        let vessel = await client.send(spaceCenter.getActiveVessel());
        console.log(vessel);
    }

    main();

I see the following error:

    internal/buffer.js:227
        throw new ERR_INVALID_ARG_TYPE('offset', 'number', offset);
        ^

    TypeError [ERR_INVALID_ARG_TYPE]: The "offset" argument must be of type number. Received type undefined
        at Buffer.readIntLE (internal/buffer.js:227:11)
        at Object.decodeUInt64 [as decode] (d:\workspace\krpc-test\node_modules\krpc-node\lib\decoders.js:67:19)
        at decodeObject (d:\workspace\krpc-test\node_modules\krpc-node\lib\client.js:476:21)
        at decode (d:\workspace\krpc-test\node_modules\krpc-node\lib\client.js:429:16)
        at d:\workspace\krpc-test\node_modules\krpc-node\lib\client.js:394:24
        at Array.map (<anonymous>)
        at onMessage (d:\workspace\krpc-test\node_modules\krpc-node\lib\client.js:388:41)
        at WebSocket.client.rpc.socket.onmessage (d:\workspace\krpc-test\node_modules\krpc-node\lib\client.js:278:9)
        at WebSocket.onMessage (d:\workspace\krpc-test\node_modules\ws\lib\event-target.js:120:16)
        at WebSocket.emit (events.js:189:13)
    Waiting for the debugger to disconnect...
    NodeError: The "offset" argument must be of type number. Received type undefined
    errors.js:158
        at Buffer.readIntLE (internal/buffer.js:227:11)
        at Object.decodeUInt64 [as decode] (d:\workspace\krpc-test\node_modules\krpc-node\lib\decoders.js:67:19)
        at decodeObject (d:\workspace\krpc-test\node_modules\krpc-node\lib\client.js:476:21)
        at decode (d:\workspace\krpc-test\node_modules\krpc-node\lib\client.js:429:16)
        at d:\workspace\krpc-test\node_modules\krpc-node\lib\client.js:394:24
        at Array.map (<anonymous>)
        at onMessage (d:\workspace\krpc-test\node_modules\krpc-node\lib\client.js:388:41)
        at WebSocket.client.rpc.socket.onmessage (d:\workspace\krpc-test\node_modules\krpc-node\lib\client.js:278:9)
        at WebSocket.onMessage (d:\workspace\krpc-test\node_modules\ws\lib\event-target.js:120:16)
        at WebSocket.emit (events.js:189:13)

**Comment by eXigentCoder** [comment]
Apologies, I have been travelling over the holiday period and taking a break from coding, but back now and will take a look ASAP, just got some life admin to take care of.

**Comment by leoworm** [comment]
Hey, sorry to necro this, but I'm having the exact same issue. I've tried using KSP version 1.5.1 and the new 1.7, both give me the same error. Did anyone ever manage to solve this?

```
internal/buffer.js:235
    throw new ERR_INVALID_ARG_TYPE('offset', 'number', offset);
    ^

TypeError [ERR_INVALID_ARG_TYPE]: The "offset" argument must be of type number. Received type undefined
    at Buffer.readIntLE (internal/buffer.js:235:11)
    at Object.decodeUInt64 [as decode] (C:\Users\leoch\Documents\KSP-Code\ksp-data-server\node_modules\krpc-node\lib\decoders.js:67:19)
    at decodeObject (C:\Users\leoch\Documents\KSP-Code\ksp-data-server\node_modules\krpc-node\lib\client.js:476:21)
    at decode (C:\Users\leoch\Documents\KSP-Code\ksp-data-server\node_modules\krpc-node\lib\client.js:429:16)
    at decodeCollection (C:\Users\leoch\Documents\KSP-Code\ksp-data-server\node_modules\krpc-node\lib\client.js:452:32)
    at decode (C:\Users\leoch\Documents\KSP-Code\ksp-data-server\node_modules\krpc-node\lib\client.js:423:16)
    at C:\Users\leoch\Documents\KSP-Code\ksp-data-server\node_modules\krpc-node\lib\client.js:394:24
    at Array.map (<anonymous>)
    at onMessage (C:\Users\leoch\Documents\KSP-Code\ksp-data-server\node_modules\krpc-node\lib\client.js:388:41)
    at WebSocket.client.rpc.socket.onmessage (C:\Users\leoch\Documents\KSP-Code\ksp-data-server\node_modules\krpc-node\lib\client.js:278:9)
```

**Comment by eXigentCoder** [comment]
Hey @leoworm and @chemuduguntar 
Sorry it took me so long, life got ... interesting. I just booted up the game (1.7.0.2483) with just the KRPC mod installed (v0.4.8) and ran the above code in an Integration test which is now checked in the code base and I successfully get back the vessel object.
I have included a screenshot of what my ingame settings looked like, make sure you are using Websockets and not TCP/IP. I had to change the port numbers because I was getting an AccessDenied error when attempting to start the server.
<img width="151" alt="2019-05-08 09_11_21-Kerbal Space Program" src="https://user-images.githubusercontent.com/8137674/57356881-df1d6700-7171-11e9-8dd4-860224a774d7.png">

Just to confirm the situation where you are getting this error:

- The game is running
- KRPC is installed
- The KRPC server is running
- The KRPC server is in websocket mode not TCP/IP
- You are in the "flight scene" i.e. you have a vessel loaded and can move around

If the above is true, would you mind checking out the source code to your local pc and running the tests locally by first typing in `npm i` and then when that is successful `npm t`

**Comment by leoworm** [comment]
Did that (with KSP running in the Websocket mode), got this

` npm t

> krpc-node@4.0.1 test C:\Users\leoch\Documents\krpc-node-master
> mocha test/**/*.js



  Bugfixes - ERR_INVALID_ARG_TYPE
    √ should retrieve the active vessel (770ms)

  Bugfixes - part finder issue
Error getting data for undefined TypeError: Cannot read property 'get' of undefined
    at logVesselData (C:\Users\leoch\Documents\krpc-node-master\test\api\bug-reproduction\part-finder.js:28:44)
    at Context.<anonymous> (C:\Users\leoch\Documents\krpc-node-master\test\api\bug-reproduction\part-finder.js:14:19)
    at processTicksAndRejections (internal/process/next_tick.js:81:5)
Uncaught Exception TypeError [ERR_INVALID_ARG_TYPE]: The "offset" argument must be of type number. Received type undefined
    at Buffer.readIntLE (internal/buffer.js:235:11)
    at Object.decodeUInt64 [as decode] (C:\Users\leoch\Documents\krpc-node-master\lib\decoders.js:67:19)
    at decodeObject (C:\Users\leoch\Documents\krpc-node-master\lib\client.js:476:21)
    at decode (C:\Users\leoch\Documents\krpc-node-master\lib\client.js:429:16)
    at decodeCollection (C:\Users\leoch\Documents\krpc-node-master\lib\client.js:452:32)
    at decode (C:\Users\leoch\Documents\krpc-node-master\lib\client.js:423:16)
    at C:\Users\leoch\Documents\krpc-node-master\lib\client.js:394:24
    at Array.map (<anonymous>)
    at onMessage (C:\Users\leoch\Documents\krpc-node-master\lib\client.js:388:41)
    at WebSocket.client.rpc.socket.onmessage (C:\Users\leoch\Documents\krpc-node-master\lib\client.js:278:9)
    at WebSocket.onMessage (C:\Users\leoch\Documents\krpc-node-master\node_modules\ws\lib\event-target.js:120:16)
    at WebSocket.emit (events.js:197:13)
    at Receiver._receiver.onmessage (C:\Users\leoch\Documents\krpc-node-master\node_modules\ws\lib\websocket.js:137:47)
    at Receiver.dataMessage (C:\Users\leoch\Documents\krpc-node-master\node_modules\ws\lib\receiver.js:397:14)
    at Receiver.getData (C:\Users\leoch\Documents\krpc-node-master\node_modules\ws\lib\receiver.js:347:12)
    at Receiver.startLoop (C:\Users\leoch\Documents\krpc-node-master\node_modules\ws\lib\receiver.js:143:16)
    at Receiver.add (C:\Users\leoch\Documents\krpc-node-master\node_modules\ws\lib\receiver.js:117:10)
    at Socket.emit (events.js:197:13)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:145:17) TypeError [ERR_INVALID_ARG_TYPE]: The "offset" argument must be of type number. Received type undefined
    at Buffer.readIntLE (internal/buffer.js:235:11)
    at Object.decodeUInt64 [as decode] (C:\Users\leoch\Documents\krpc-node-master\lib\decoders.js:67:19)
    at decodeObject (C:\Users\leoch\Documents\krpc-node-master\lib\client.js:476:21)
    at decode (C:\Users\leoch\Documents\krpc-node-master\lib\client.js:429:16)
    at decodeCollection (C:\Users\leoch\Documents\krpc-node-master\lib\client.js:452:32)
    at decode (C:\Users\leoch\Documents\krpc-node-master\lib\client.js:423:16)
    at C:\Users\leoch\Documents\krpc-node-master\lib\client.js:394:24
    at Array.map (<anonymous>)
    at onMessage (C:\Users\leoch\Documents\krpc-node-master\lib\client.js:388:41)
    at WebSocket.client.rpc.socket.onmessage (C:\Users\leoch\Documents\krpc-node-master\lib\client.js:278:9)
    at WebSocket.onMessage (C:\Users\leoch\Documents\krpc-node-master\node_modules\ws\lib\event-target.js:120:16)
    at WebSocket.emit (events.js:197:13)
    at Receiver._receiver.onmessage (C:\Users\leoch\Documents\krpc-node-master\node_modules\ws\lib\websocket.js:137:47)
    at Receiver.dataMessage (C:\Users\leoch\Documents\krpc-node-master\node_modules\ws\lib\receiver.js:397:14)
    at Receiver.getData (C:\Users\leoch\Documents\krpc-node-master\node_modules\ws\lib\receiver.js:347:12)
    at Receiver.startLoop (C:\Users\leoch\Documents\krpc-node-master\node_modules\ws\lib\receiver.js:143:16)
    at Receiver.add (C:\Users\leoch\Documents\krpc-node-master\node_modules\ws\lib\receiver.js:117:10)
    at Socket.emit (events.js:197:13)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:145:17)
C:\Users\leoch\Documents\krpc-node-master\test\init.js:13
    throw err;
    ^

TypeError [ERR_INVALID_ARG_TYPE]: The "offset" argument must be of type number. Received type undefined
    at Buffer.readIntLE (internal/buffer.js:235:11)
    at Object.decodeUInt64 [as decode] (C:\Users\leoch\Documents\krpc-node-master\lib\decoders.js:67:19)
    at decodeObject (C:\Users\leoch\Documents\krpc-node-master\lib\client.js:476:21)
    at decode (C:\Users\leoch\Documents\krpc-node-master\lib\client.js:429:16)
    at decodeCollection (C:\Users\leoch\Documents\krpc-node-master\lib\client.js:452:32)
    at decode (C:\Users\leoch\Documents\krpc-node-master\lib\client.js:423:16)
    at C:\Users\leoch\Documents\krpc-node-master\lib\client.js:394:24
    at Array.map (<anonymous>)
    at onMessage (C:\Users\leoch\Documents\krpc-node-master\lib\client.js:388:41)
    at WebSocket.client.rpc.socket.onmessage (C:\Users\leoch\Documents\krpc-node-master\lib\client.js:278:9)
    at WebSocket.onMessage (C:\Users\leoch\Documents\krpc-node-master\node_modules\ws\lib\event-target.js:120:16)
    at WebSocket.emit (events.js:197:13)
    at Receiver._receiver.onmessage (C:\Users\leoch\Documents\krpc-node-master\node_modules\ws\lib\websocket.js:137:47)
    at Receiver.dataMessage (C:\Users\leoch\Documents\krpc-node-master\node_modules\ws\lib\receiver.js:397:14)
    at Receiver.getData (C:\Users\leoch\Documents\krpc-node-master\node_modules\ws\lib\receiver.js:347:12)
    at Receiver.startLoop (C:\Users\leoch\Documents\krpc-node-master\node_modules\ws\lib\receiver.js:143:16)
    at Receiver.add (C:\Users\leoch\Documents\krpc-node-master\node_modules\ws\lib\receiver.js:117:10)
    at Socket.emit (events.js:197:13)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:145:17)
npm ERR! Test failed.  See above for more details.`

This is really weird...

**Comment by leoworm** [comment]
Ok, so I just did the test again, this time on my linux machine. Here it worked without issues. So it seems the problem only appears on Windows.

**Comment by eXigentCoder** [comment]
Have merged in #17 which should fix this and done a publish to npm. Going to close this issue now, please feel free to reopen if it isn't actually fixed.
